### PR TITLE
Ben pantheon urls resolve merge conflicts

### PR DESCRIPTION
**Pull request by xdavidson**
This PR is based off https://github.com/redhataccess/pantheon/pull/422. It resolves the merge conflicts from master.

### CODE REVIEW

File: pantheon-bundle/src/main/java/com/redhat/pantheon/extension/url/CustomerPortalUrlProvider.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.redhat.pantheon.extension.url;
+
+import com.redhat.pantheon.helper.PantheonConstants;
+import com.redhat.pantheon.model.document.DocumentVariant;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+
+import java.util.Optional;
+
+public abstract class CustomerPortalUrlProvider implements UrlProvider {
```
Comment (carlosmunoz) [inline pantheon-bundle/src/main/java/com/redhat/pantheon/extension/url/CustomerPortalUrlProvider.java:10]: Add class-level comments

File: pantheon-bundle/src/main/java/com/redhat/pantheon/extension/url/CustomerPortalUrlUuidProvider.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import com.redhat.pantheon.model.ProductVersion;
+import com.redhat.pantheon.model.document.DocumentMetadata;
+import com.redhat.pantheon.model.document.DocumentVariant;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.RepositoryException;
+
+public class CustomerPortalUrlUuidProvider extends CustomerPortalUrlProvider {
```
Comment (carlosmunoz) [inline pantheon-bundle/src/main/java/com/redhat/pantheon/extension/url/CustomerPortalUrlUuidProvider.java:11]: Add a class-level comment

**Review by carlosmunoz — APPROVED** [review]
Simple documentation improvements, but since Lisa is not the original author we are not concentrating on deep changes.

File: pantheon-bundle/src/main/java/com/redhat/pantheon/extension/url/CustomerPortalUrlProvider.java (hunk 1 above)
Comment (xdavidson) [reply pantheon-bundle/src/main/java/com/redhat/pantheon/extension/url/CustomerPortalUrlProvider.java:18]: done

File: pantheon-bundle/src/main/java/com/redhat/pantheon/extension/url/CustomerPortalUrlUuidProvider.java (hunk 2 above)
Comment (xdavidson) [reply pantheon-bundle/src/main/java/com/redhat/pantheon/extension/url/CustomerPortalUrlUuidProvider.java:16]: done

**Review by carlosmunoz — APPROVED** [review]

**Pull request merged**